src/mds/MDSRank.cc

```diff
@@ -2130,6 +2130,8 @@ void MDSRank::active_start()
 {
   dout(1) << "active_start" << dendl;
 
+  m_is_active = true;
```

Comment:
I don't see where you are resetting it. I'd suggest that you put the code at the end of `handle_mds_map`:

``` 
    m_is_active = is_active();
```

Reply:
resetting?

Reply:
IMO no need to reset it.

@batrick  BTW, why not just check the MDS' state from the `mdsmap` instead of adding a new `m_is_active` ? For lockless case ?

Reply:
You need the `mds_lock` to look at the MDSMap. We don't want the metrics aggregator to be acquiring that lock generally.

Reply:
Makes sense.